### PR TITLE
🗃️ Add Participant Prisma schema to PI DAS

### PR DESCRIPTION
**Pull request by alyssadsouza**
- added the `Participant` Prisma schema and seed data
- added create + read + (general) update operations
- added the `/participants` router + swagger jsdocs

### CODE REVIEW

File: apps/pi-das/prisma/migrations/20230927144655_participant/migration.sql [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+-- CreateTable
+CREATE TABLE "Participant" (
+  "id" TEXT NOT NULL,
+  "dateOfBirth" TIMESTAMP(3) NOT NULL,
+  "emailAddress" TEXT NOT NULL,
+  "guardianEmailAddress" TEXT,
+  "guardianName" TEXT,
+  "guardianPhoneNumber" CHAR(10),
+  "guardianRelationship" TEXT,
+  "inviteId" TEXT,
```
Comment (alyssadsouza) [inline apps/pi-das/prisma/migrations/20230927144655_participant/migration.sql:13]: note that this isn't documented in the Data Model tables on the Wiki, but is included in v3 of Hardeep's spreadsheet as optional so I've added it here. the foreign key constraint is added in #126

File: apps/pi-das/prisma/migrations/20230927144655_participant/migration.sql [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  "mailingAddressCity" TEXT,
+  "mailingAddressPostalCode" CHAR(6),
+  "mailingAddressProvince" TEXT,
```
Comment (alyssadsouza) [inline apps/pi-das/prisma/migrations/20230927144655_participant/migration.sql:16]: will update this to be a dropdown (Enum type)

File: apps/pi-das/prisma/migrations/20230927144655_participant/migration.sql [hunk 3]
```diff
@@ -0,0 +1,29 @@
+-- DropTable
+DROP TABLE "Participant";
+
+-- CreateTable
+CREATE TABLE "Participant" (
+  "id" TEXT NOT NULL,
+  "dateOfBirth" TIMESTAMP(3) NOT NULL,
```
Comment (alyssadsouza) [inline apps/pi-das/prisma/migrations/20230927144655_participant/migration.sql:7]: considering updating from `TIMESTAMP(3)` to `DATE` (by [adding `@db.Date` property](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#postgresql-6) in Prisma schema)

**Review by mistryrn — APPROVED** [review]
Looks good! Nice details in the seed data 😂 Thanks for updating for the ESM stuff too!

**Pull request merged**